### PR TITLE
meson.build: disable debugger on linux/s390x because there's no support

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -237,12 +237,14 @@ else
   message('Using bundled openssl code')
 endif
 
+has_debugger = not (host_machine.system() == 'linux' and host_machine.cpu_family() == 's390x')
```
Comment (radare) [inline meson.build]: can we move this into a separate function, there will be more cases that we dont support the debugger. But in theory, r2 should build in those platforms, but the debug plugin should do nothing and warn at runtime.

File: meson.build [hunk 2]
```diff
@@ -237,12 +237,14 @@ else
...
+
 userconf = configuration_data()
 userconf.set('HAVE_LIB_MAGIC', sys_magic.found().to_int())
 userconf.set('USE_LIB_MAGIC', use_syslib_magic.to_int())
 userconf.set('HAVE_LIB_XXHASH', sys_xxhash.found().to_int())
 userconf.set('USE_LIB_XXHASH', use_sys_xxhash.to_int())
-userconf.set('DEBUGGER', 1)
+userconf.set('DEBUGGER', has_debugger.to_int())
```
Comment (radare) [inline meson.build]: just call a function that returns 0|1 as int

File: meson.build (hunk 1 above)
Comment (ret2libc) [reply meson.build]: https://mesonbuild.com/Syntax.html#userdefined-functions-and-methods it's not possible, sorry. This is why I have defined this variable. I agree that there should be an error at runtime though. We should have an "empty" debug plugin or something.

File: meson.build (hunk 2 above)
Comment (ret2libc) [reply meson.build]: see above.

**Comment by radare** [comment]
Then i prefer to have the fix in the c code rather than this check which is incomplete and only in meson. Because acr doesnt checks the same

**Comment by ret2libc** [comment]
ok

**Pull request closed without merging**